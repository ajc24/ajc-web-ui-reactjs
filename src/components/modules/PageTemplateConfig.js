/**
 * Developed by Anthony Cox in 2025
 */
import { storybookCssClassList } from '../data/storybook';

/* Commonly used CSS classes */
const bodyDefaultCss = 'body-default';

/**
 * Functionality designed to work with the page template components (headers, menu bars, body content, footers)
 */
export default class PageTemplateConfig {
  /**
   * Setup and configure the current web page to ensure the default CSS styling is set to it.
   * This also handles the styling of pages which are deployed on Storybook
   */
  static setupDocumentBodyCss() {
    /* If we are rendering on Storybook, remove the Storybook classes which will affect the width of the component */
    for (let index = 0; index < storybookCssClassList.length; index += 1) {
      if (document.body.classList.contains(storybookCssClassList[index]) === true) {
        document.body.classList.remove(storybookCssClassList[index]);
      }
    }
    /* Ensure that the <body> element stretches to 100% of the width of the screen, that the X axis scrollbar is hidden and Y axis scrollbar is always displayed */
    if (document.body.classList.contains(bodyDefaultCss) === false) {
      document.body.classList.add(bodyDefaultCss);
      document.activeElement.blur();
    }
  }
}
