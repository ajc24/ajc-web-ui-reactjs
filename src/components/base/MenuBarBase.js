/**
 * Developed by Anthony Cox in 2025
 */
import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import PageTemplateConfig from '../modules/PageTemplateConfig';
import '../css/common.css';
import './css/base-menu-bar.css';

const defaultComponentId = 'default-id--menu-bar-base';

/**
 * Baseline component setting the template for all Menu Bar components to be developed from.
 * This baseline auto-handles screen widths from the most commonly used mobile screen sizes (360x800) to the most commonly used desktop sizes (1920x1080).
 * The height of the component is 55px.
 */
const MenuBarBase = props => {
  const [id, setId] = useState(defaultComponentId);

  useEffect(() => {
    /* Setup the default CSS styling for the page which is rendering this component */
    PageTemplateConfig.setupDocumentBodyCss();

    /* Set the ID for the component */
    if (props.id !== undefined) {
      setId(`${props.id}--menu-bar-base`);
    }
  });

  /* Set the styling for the navigation element */
  let navCss = 'nav screen-width-root background-white';
  if (props.backgroundColour === 'gold' || props.backgroundColour === 'green-2') {
    navCss += ' font-black';
  } else if (props.backgroundColour === 'navy-and-gold') {
    navCss += ' font-gold';
  } else {
    navCss += ' font-white';
  }
  /* Set the styling for the outer content element */
  let outerContentCss = 'menu-bar-content-outer screen-width-content-outer';
  if (props.backgroundColour === 'gold' || props.backgroundColour === 'green' || props.backgroundColour === 'green-2' || props.backgroundColour === 'red'
    || props.backgroundColour === 'white') {
      /* Set the background colour to the specified colour */
      outerContentCss += ` background-${props.backgroundColour}`;
  } else if (props.backgroundColour === 'grey') {
    /* Set the background colour as grey */
    outerContentCss += ` background-${props.backgroundColour}-2`;
  } else if (props.backgroundColour === 'navy-and-gold' || props.backgroundColour === 'navy-and-white') {
    /* Set the background colour as navy */
    outerContentCss += ' background-navy';
  } else {
    /* Default set the background colour as white */
    outerContentCss += ` background-white`;
  }
  /* Set the styling for the inner content element */
  const innerContentCss = 'menu-bar-content-inner screen-width-content-inner';
  return (
    <nav id={id} className={navCss}>
      <div className={outerContentCss}>
        <div className={innerContentCss}>
          {props.children}
        </div>
      </div>
    </nav>
  );
}
MenuBarBase.propTypes = {
  /** The background colour for the menu bar. The default colour for the background is white. */
  backgroundColour: PropTypes.oneOf([ 'green', 'green-2', 'grey', 'navy-and-gold', 'navy-and-white', 'red', 'white' ]),
  /** The content to be displayed within the menu bar template component. */
  children: PropTypes.any,
  /** The unique identifier for this component. */
  id: PropTypes.string,
};
export default MenuBarBase;
