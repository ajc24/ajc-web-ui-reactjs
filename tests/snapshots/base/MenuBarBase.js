/**
 * Developed by Anthony Cox in 2024
 */
import 'jsdom-global/register';
import React from 'react';
import { TestDev } from 'ajc-testing-code';
import MenuBarBase from '../../../src/components/base/MenuBarBase';
import PageTemplateConfig from '../../../src/components/modules/PageTemplateConfig';

describe('Menu Bar Base', () => {
  /* Set the spies for use in the tests */
  let setupDocumentBodyCssSpy;

  beforeAll(() => {
    setupDocumentBodyCssSpy = jest
      .spyOn(PageTemplateConfig, 'setupDocumentBodyCss')
      .mockImplementation(() => {});
  });

  afterAll(() => {
    setupDocumentBodyCssSpy.mockRestore();
  });

  describe('Component with default id', () => {
    let snapshot;
    
    beforeAll(() => {
      snapshot = TestDev.createSnapshot(
        <React.Fragment>
          <MenuBarBase>
            Menu bar base text content.
          </MenuBarBase>
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('Component with custom id', () => {
    let snapshot;
    
    beforeAll(() => {
      snapshot = TestDev.createSnapshot(
        <React.Fragment>
          <MenuBarBase id="custom-id">
            Menu bar base text content.
          </MenuBarBase>
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(snapshot).toMatchSnapshot();
    });
  });
});
