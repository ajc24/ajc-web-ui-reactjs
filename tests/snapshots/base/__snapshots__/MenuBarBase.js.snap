// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Bar Base Component with custom id verifies the snapshot for the component 1`] = `"<nav id="custom-id--menu-bar-container" class="menu-bar-container background-white"><div class="menu-bar-content-container background-transparent">Menu bar base text content.</div></nav>"`;

exports[`Menu Bar Base Component with default id verifies the snapshot for the component 1`] = `"<nav id="default-id--menu-bar-container" class="menu-bar-container background-white"><div class="menu-bar-content-container background-transparent">Menu bar base text content.</div></nav>"`;
